src/connector/nodejs/test/cases/test.stmt.js

```diff
@@ -0,0 +1,915 @@
+let tagData1 = [true, 1, 32767, 1234555, -164243520000011111n, 214.02, 2.01, 'taosdata涛思数据', 'TDengine数据', 254, 65534, 4294967290 / 2, 164243520000011111n];
+let tagData2 = [true, 2, 32767, 1234555, -164243520000011111n, 214.02, 2.01, 'taosdata涛思数据', 'TDengine数据', 254, 65534, 4294967290 / 2, 164243520000011111n];
+let tagData3 = [true, 3, 32767, 1234555, -164243520000011111n, 214.02, 2.01, 'taosdata涛思数据', 'TDengine数据', 254, 65534, 4294967290 / 2, 164243520000011111n];
+
+/**
+ * Combine individual array of every tdengine type that 
+ * has been declared and then return a new array.
+ * @returns return data array.
+ */
+function getBindData() {
+  let bindDataArr = [];
+  for (let i = 0; i < 5; i++) {
+    bindDataArr.push(tsArr[i]);
+    bindDataArr.push(boolArr[i]);
+    bindDataArr.push(tinyIntArr[i]);
+    bindDataArr.push(smallIntArr[i]);
+    bindDataArr.push(intArr[i]);
+    bindDataArr.push(bigIntArr[i]);
+    bindDataArr.push(floatArr[i]);
+    bindDataArr.push(doubleArr[i]);
+    bindDataArr.push(binaryArr[i]);
+    bindDataArr.push(ncharArr[i]);
+    bindDataArr.push(uTinyIntArr[i]);
+    bindDataArr.push(uSmallIntArr[i]);
+    bindDataArr.push(uIntArr[i]);
+    bindDataArr.push(uBigIntArr[i]);
+  }
+  return bindDataArr;
+}
+
+function executeUpdate(sql) {
+  console.log(sql);
+  c1.execute(sql);
+}
+
+function executeQuery(sql) {
+  c1.execute(sql, { quiet: true })
+  var data = c1.fetchall();
+  let fields = c1.fields;
+  let resArr = [];
+
+  data.forEach(row => {
+    row.forEach(data => {
+      if (data instanceof Date) {
+        // console.log("date obejct:"+data.valueOf());
+        resArr.push(data.taosTimestamp());
+      } else {
+        // console.log("not date:"+data);
+        resArr.push(data);
+      }
+      // console.log(data instanceof Date)
+    })
+  })
+  return { resData: resArr, resFeilds: fields };
+}
+
+beforeAll(() => {
+  conn = taos.connect({ host: "127.0.0.1", user: "root", password: "taosdata", config: "/etc/taos", port: 10 });
+  c1 = conn.cursor();
+  executeUpdate(`create database if not exists ${dbName} keep 3650;`);
+  executeUpdate(`use ${dbName};`);
+});
+
+// Clears the database and adds some testing data.
+// Jest will wait for this promise to resolve before running tests.
+afterAll(() => {
+  // executeUpdate(`drop database if exists ${dbName};`);
+  conn.close();
+});
+
+describe("stmt_bind_single_param", () => {
+  test(`name:bindSingleParamWithOneTable;` +
+    `author:${author};` +
+    `desc:Using stmtBindSingleParam() bind one table in a batch;` +
+    `filename:${fileName};` +
+    `result:${result}`, () => {
+      let table = 'bindsingleparambatch_121';
+      let createSql = `create table if not exists ${table} ` +
+        `(ts timestamp,` +
+        `bl bool,` +
+        `i8 tinyint,` +
+        `i16 smallint,` +
+        `i32 int,` +
+        `i64 bigint,` +
+        `f32 float,` +
+        `d64 double,` +
+        `bnr binary(20),` +
+        `blob nchar(20),` +
```

Comment:
3.0 will have a new type blob, we should not use it here to avoid rename it for 3.0.